app/views/unattended/provisioning_templates/snippet/container_certs_setup.erb

```diff
@@ -0,0 +1,51 @@
+<%#
+kind: snippet
+name: container_certs_setup
+model: ProvisioningTemplate
+snippet: true
+description: |
+  This snippet sets up SSL certificates for secure communication with the Foreman registry.
+-%>
+container_certs_setup() {
+  echo "#"
+  echo "Running cert setup for container registry"
+  echo "#"
+
+  # Extract server hostname from @registration_url
+  hostname=$(echo "<%= @registration_url %>" | sed -e 's|https\?://||' -e 's|/.*||')
```

Comment:
Is there a way we can get the host directly? Not sure where `@registration_url` is coming from, but SmartProxy also has a `registration_host` method..

Reply:
Ack..that makes sense..We can rely on registration_url to provide the host directly using URI instead of string parsing inline. Not immediately clear to me why we need to worry about registration_host from smart_proxy? Is that for load_balanced setups? I haven't tested cert_auth on load balanced setups so I am not a 100% sure what URL we need to provide here. I can test that outside the scope of this PR.

Reply:
> Not immediately clear to me why we need to worry about registration_host from smart_proxy? Is that for load_balanced setups?

Exactly. It's used when setting host content source during registration, to compare with the list of known smart proxies.